akka-remote/src/main/scala/akka/remote/artery/FlightRecorder.scala

```diff
@@ -274,7 +282,8 @@ private[akka] class FlightRecorder(val fileChannel: FileChannel) extends AtomicB
     // Coalesce concurrent snapshot requests into one, i.e. ignore the "late-comers".
     // In other words, this is a critical section in which participants either enter, or just
     // simply skip ("Hm, seems someone else already does it. ¯\_(ツ)_/¯ ")
-    if (!get && compareAndSet(false, true)) {
+    val snapshotLatch = new CountDownLatch(1)
+    if (!get.isInstanceOf[SnapshotInProgress] && compareAndSet(Running, SnapshotInProgress(snapshotLatch))) {
```

Comment:
yes, it avoids a CAS in case there is already a snapshot going on. Snapshots are not that common though so this might be just premature optimization